src/System.Management.Automation/FormatAndOutput/common/Utilities/Mshexpression.cs

```diff
@@ -123,13 +125,20 @@ internal List<MshExpression> ResolveNames(PSObject target, bool expand)
             {
                 // get the members first: this will expand the globbing on each parameter
                 members = target.Members.Match(_stringValue,
-                                            PSMemberTypes.Properties | PSMemberTypes.PropertySet);
+                                            PSMemberTypes.Properties | PSMemberTypes.PropertySet | PSMemberTypes.Dynamic);
             }
             else
             {
                 // we have no globbing: try an exact match, because this is quicker.
                 PSMemberInfo x = target.Members[_stringValue];
 
+                if ((x == null) && (target.BaseObject is System.Dynamic.IDynamicMetaObjectProvider))
+                {
+                    // We could check if GetDynamicMemberNames includes the name...  but
```

Comment:
About this comment, we have already checked `GetDynamicMemberNames` when reaching here (`target.Members` includes the dynamic members returned from `GetDynamicMemberNames`). So maybe we can change comment to: `GetDynamicMemberNames does not return this name, but GetDynamicMemberNames is only a hint ...`

Reply:
Your comment is not quite correct. (Maybe it *should* be correct and there is another bug; I'm not sure what the intended behavior is.)

If you iterate through `.Members` you will indeed see the dynamic members. However, if you do `.Members[propName]`, where `propName` is the name of one of those dynamic members... you will get `null` back.

Sample output:

```powershell
PS C:\src\PowerShell> $myObj.PSObject.Members | where Name -eq FooProp


MemberType      : Dynamic
Value           :
TypeNameOfValue : dynamic
Name            : FooProp
IsInstance      : True



PS C:\src\PowerShell> $myObj.PSObject.Members['FooProp']
PS C:\src\PowerShell> $myObj.PSObject.Members['FooProp'] -eq $null
True
PS C:\src\PowerShell>
```

I'm not sure if that's a bug or By Design. Because you can't do much with PSDynamicMembers (for instance, the getter/setter just throws), maybe it is intentionally excluded when indexing directly in. But it does seem odd.

Reply:
I see. That does seem odd. I will look into it tomorrow. Thanks for bringing it up!.